src/contracts.rs

```diff
@@ -111,8 +113,18 @@ pub fn read_locktime_from_contract(redeemscript: &Script) -> i64 {
     //note that locktimes in these contracts are always 2 bytes
     //so for example a locktime of 30 wont work, as the specific single opcode
     //will be used instead
-    let rs = redeemscript.to_bytes();
-    rs[100] as i64 | (rs[101] as i64) << 8
+    let mut rs = Cursor::new(redeemscript.to_bytes());
+
+    rs.set_position(99);
+
+    let locktime_byte_length = rs.read_u8().unwrap() as usize;
+
+    let mut locktime_buff = vec![0u8; std::mem::size_of::<i64>()];
+    for i in 0..locktime_byte_length {
+        locktime_buff[i] = rs.read_u8().unwrap();
+    }
+
+    i64::from_le_bytes(locktime_buff.try_into().unwrap())
```

Comment:
I don't see the point of rewriting this function? Also I'm pretty sure rust-bitcoin has a function somewhere already which reads varints, you've just reimplemented it here.

Reply:
Yeah I have looked it in up rust-lightning and found they are encoding the time as `i64` as well. Reverted with the latest update. 

I do have a doubt though. 
The comment above the code says locktime in these contact should always be 2 bytes. I suppose its for 16 bit encoding of the locktime value by BIP68. 

In that case wont it be useful to enforce that restriction in the API somehow? Maybe use an `u16` for locktime value here?
https://github.com/bitcoin-teleport/teleport-transactions/blob/4ca8b7f49aa1b4eeab693adfa7f8b5caaa24f16c/src/contracts.rs#L75-L82

Reply:
OK change it to u16 if you like, but in a different PR.